lib/deep_merge.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+function deepMerge(target, source) {
+  target = Object.assign({}, target);
+  source = Object.assign({}, source);
```

Comment:
need to be removed, otherwise

``` js
const obj1 = {a: {b: 1, c: 1, d: {e: 1, f: 1}}};
const obj2 = {a: {b: 2, d: {f: 'f'} }};

// { a: { b: 1, c: 1, d: { e: 1, f: 1 } } }
// f should be 'f'
```

Reply:
https://github.com/hexojs/hexo-util/blob/d652ff12092c96b34908a90a2207731bbba30865/test/deep_merge.spec.js#L34

Since the unit test passed, I believe there is no problem here.

Reply:
The unit test didn't run, I suggest not to rely on it entirely. I tested locally and can verify that currently `f` key has value of `1` not `'f'`.